### ISSUE TITLE
compiler panic on invalid procedure aliasing

### ISSUE THREAD

**Issue by awwdev** [odin-lang/Odin#1392]
## Context
Odin: `dev-2021-12:eec61c3f`
The compiler flag `check` will not produce the panic. You need to use `build` (or `run`).

## Current Behavior
The following invalid syntax causes a compiler panic:
```odin
FOO :: proc() //same outcome with or without the attribute #type 
foo :: FOO {} //invalid syntax (Odin does not have this type of procedure aliasing)
foo()
```
`Odin\src\llvm_backend_general.cpp(2409): Panic: nullptr value for expression from identifier: <your_package_name>.foo (29d1538d140) : FOO @ 29cbffa1b90`

## Expected Behavior
Instead of panicking, the compiler should report an error message about this invalid syntax. Additionally, the parser alone (invoked by `check`) should already catch the error, prior to compilation.

**Comment by gingerBill** [comment]
`FOO {} //invalid syntax`
Is actually valid syntax, it's just defining a `nil` procedure of `FOO` type

However, this is a bug in the backend.

**Comment by awwdev** [comment]
Appears to be fixed (tested with `version dev-2022-09:98420192`), there is no more compiler panic.
It reports the following user error instead:
`Illegal declaration of a constant procedure value`
